include/circt/Firtool/Firtool.h

```diff
@@ -0,0 +1,246 @@
+      llvm::cl::cat(category)};
+
+  llvm::cl::opt<bool> enableAnnotationWarning{
+      "warn-on-unprocessed-annotations",
+      llvm::cl::desc(
+          "Warn about annotations that were not removed by lower-to-hw"),
+      llvm::cl::init(false), llvm::cl::cat(category)};
+
+  llvm::cl::opt<bool> addMuxPragmas{
+      "add-mux-pragmas", llvm::cl::desc("Annotate mux pragmas"),
+      llvm::cl::init(false), llvm::cl::cat(category)};
+
+  llvm::cl::opt<bool> emitChiselAssertsAsSVA{
+      "emit-chisel-asserts-as-sva",
+      llvm::cl::desc("Convert all chisel asserts into SVA"),
+      llvm::cl::init(false), llvm::cl::cat(category)};
+
+  llvm::cl::opt<bool> emitSeparateAlwaysBlocks{
+      "emit-separate-always-blocks",
+      llvm::cl::desc(
+          "Prevent always blocks from being merged and emit constructs into "
+          "separate always blocks whenever possible"),
+      llvm::cl::init(false), llvm::cl::cat(category)};
+
+  llvm::cl::opt<bool> etcDisableInstanceExtraction{
+      "etc-disable-instance-extraction",
+      llvm::cl::desc("Disable extracting instances only that feed test code"),
+      llvm::cl::init(false), llvm::cl::cat(category)};
+
+  llvm::cl::opt<bool> etcDisableModuleInlining{
+      "etc-disable-module-inlining",
+      llvm::cl::desc("Disable inlining modules that only feed test code"),
+      llvm::cl::init(false), llvm::cl::cat(category)};
+
+  llvm::cl::opt<bool> addVivadoRAMAddressConflictSynthesisBugWorkaround{
+      "add-vivado-ram-address-conflict-synthesis-bug-workaround",
+      llvm::cl::desc("Add a vivado specific SV attribute (* ram_style = "
+                     "\"distributed\" *) to array registers as a workaronud "
+                     "for a vivado synthesis bug that incorrectly modifies "
+                     "address conflict behavivor of combinational memories"),
+      llvm::cl::init(false), llvm::cl::cat(category)};
+
+  bool isRandomEnabled(RandomKind kind) const {
+    return disableRandom != RandomKind::All && disableRandom != kind;
+  }
+
+  firrtl::PreserveValues::PreserveMode getPreserveMode() const {
+    if (!buildMode.getNumOccurrences())
+      return preserveMode;
+    switch (buildMode) {
+    case BuildModeDebug:
+      return firrtl::PreserveValues::Named;
+    case BuildModeRelease:
+      return firrtl::PreserveValues::None;
+    }
+    llvm_unreachable("unknown build mode");
+  }
+
+  FirtoolOptions(llvm::cl::OptionCategory &category) : category(category) {}
+};
+
+LogicalResult populateCHIRRTLToLowFIRRTL(mlir::PassManager &pm,
+                                         const FirtoolOptions &opt,
+                                         ModuleOp module,
+                                         StringRef inputFilename);
+
+LogicalResult populateLowFIRRTLToHW(mlir::PassManager &pm,
+                                    const FirtoolOptions &opt);
+
+LogicalResult populateHWToSV(mlir::PassManager &pm, const FirtoolOptions &opt);
```

Comment:
Should we put these into a `namespace firtool`? Especially the `populateHWToSV` function sounds very general, but is firtool-specific. Might collide with some other usage in CIRCT where a non-FIRRTL tool would want to gov from HW to SV.